docker-compose.yml.template

```diff
@@ -0,0 +1,28 @@
+volumes:
+  overpass-db:
+
+version: '3.8'
+services:
+  overpass:
+    # specify the image here
+    image: wiktorn/overpass-api:0.7.56.9
+    container_name: overpass
+    # uncomment if you want to build the image yourself
+    # build: .
+    ports:
+      - 12345:80
+    volumes:
+      # use a docker managed volume
+      - overpass-db:/db
+    environment:
+      - OVERPASS_META=no
+      - OVERPASS_MODE=init
+      - OVERPASS_PLANET_URL=http://download.geofabrik.de/europe/monaco-latest.osm.bz2
+      - OVERPASS_DIFF_URL=http://download.openstreetmap.fr/replication/europe/monaco/minute/
+      - OVERPASS_UPDATE_SLEEP=3600
+      - OVERPASS_USE_AREAS=false
+    healthcheck:
+      test: ["CMD-SHELL", "curl -f http://localhost:12345/api/status || exit 1"]
+      interval: 5m
+      start_period: 1m
+      retries: 2880  # 2 days in 5 mins intervals
```

Comment:
to avoid porting it to every single Dockerfile.. ah, just seeing the `update.py` script, nice work:) now I understand the reason for your project structure. so do you want to run that again and include the healthcheck directive in the `Dockerfile.template`? probably safer than in docker-compose, which not everyone might use?

Reply:
I don't quite follow proposed configuration for healthcheck. If I'm [reading](https://docs.docker.com/engine/reference/builder/#healthcheck) correctly, this definitions means:
- if for first minute healthcheck fails, it's not an issue
- if it fails, try again for 2 days, to consider the container "unhealthy"

I think what we want is actually reverse. Because the healthcheck will fail during initialization, we want to give it a lot of time there, until it will consider container unhealthy and killed. But after it has started, it should quickly detect that status doesn't reply and restart the container.

Moving that to Dockerfile.template is good direction. I think that as I'm already building the images myself (as DockerHub builds got really unrealiable), I could also remove the other files, and generate them during build.

Reply:
I changed it slightly just now, there were some issues.

right, I interpret it similar: 

it starts after 5 mins to do healthchecks, if curl fails after that, it will try `retries` times in `interval` intervals (in this case 2 days) _before_ considering the container unhealthy, only if it still fails after that, it's considered unhealthy and killed. We an give it even more retries if you want? While it's re-trying for 2 days it will simply say `(health: starting)`, e.g. here from my test just now:
```
CONTAINER ID        IMAGE                                                  COMMAND                  CREATED             STATUS                            PORTS                    NAMES
c2c6f30a4f1a        ghcr.io/gis-ops/overpass:0.7.55                        "/docker-entrypoint.…"   12 minutes ago      Up 3 seconds (health: starting)   0.0.0.0:31942->80/tcp    overpass
```

when it first has a successful curl it'll stop the healthcheck and report a `healthy` status:

```
CONTAINER ID        IMAGE                                                  COMMAND                  CREATED             STATUS                            PORTS                    NAMES
c2c6f30a4f1a        ghcr.io/gis-ops/overpass:0.7.55                        "/docker-entrypoint.…"   17 minutes ago      Up 5 minutes (healthy)   0.0.0.0:31942->80/tcp    overpass
```